clients/nodejs/validator-registration.js

```diff
@@ -0,0 +1,354 @@
+
+    const clientConfig = clientConfigBuilder.build();
+    const networkConfig = clientConfig.networkConfig;
+
+    $.consensus = await (!config.volatile
+        ? Nimiq.Consensus.nano(networkConfig)
+        : Nimiq.Consensus.volatileNano(networkConfig));
+
+    $.client = new Nimiq.Client(clientConfig, $.consensus);
+    $.blockchain = $.consensus.blockchain;
+    $.accounts = $.blockchain.accounts;
+    $.mempool = $.consensus.mempool;
+    $.network = $.consensus.network;
+
+    Nimiq.Log.i(TAG, `Peer address: ${networkConfig.peerAddress.toString()} - public key: ${networkConfig.keyPair.publicKey.toHex()}`);
+
+    // This is the hardcoded address dedicated to validator registration
+    const recipientAddr = Nimiq.Address.fromUserFriendlyAddress('NQ07 0000 0000 0000 0000 0000 0000 0000 0000');
+
+    // Extract the validator configuration (we read the private keys from the config file)
+    let validatorPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.ValidatorAccount.PrivateKey));
+    let signingPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.SigningKey.PrivateKey));
+    let votingSecretKey =  NimiqPOS.BLSSecretKey.fromHex(validator_config.VotingKey.SecretKey);
+
+    // Create the KeyPairs
+    const validatorKeyPair = Nimiq.KeyPair.derive(validatorPrivateKey);
+    const SigningKeyPair = Nimiq.KeyPair.derive(signingPrivateKey);
+    const votingKeyPair = NimiqPOS.BLSKeyPair.derive(votingSecretKey);
+
+    // Create a new validator registry
+    let validator = new ValidatorRegistry(validatorKeyPair.publicKey.toAddress(), SigningKeyPair.publicKey, votingKeyPair.publicKey);
+
+    // Get the transaction's data
+    let data = validator.get_serialized_data();
+
+    // We monitor the status of the validator registration transactions
+    $.client.addTransactionListener((tx) => {
+        console.log(' Transaction update: ');
+        console.log(tx);
+    }, [recipientAddr]);
+
+    let consensusState = Nimiq.Client.ConsensusState.CONNECTING;
+    $.client.addConsensusChangedListener(async (state) => {
+        consensusState = state;
+        if (state === Nimiq.Client.ConsensusState.ESTABLISHED) {
+            Nimiq.Log.i(TAG, `Blockchain ${config.type}-consensus established in ${(Date.now() - START) / 1000}s.`);
+            const chainHeight = await $.client.getHeadHeight();
+            const chainHeadHash = await $.client.getHeadHash();
+            Nimiq.Log.i(TAG, `Current state: height=${chainHeight}, headHash=${chainHeadHash}`);
+
+            const account = await $.client.getAccount(validatorKeyPair.publicKey.toAddress()).catch(() => null);
+            const balance = Nimiq.Policy.lunasToCoins(account.balance)
+
+            Nimiq.Log.i(TAG, `Validator address ${validatorKeyPair.publicKey.toAddress().toUserFriendlyAddress()}.`
+            + (account ? ` Balance: ${balance} NIM` : ''));
+
+            if (balance < 6){
```

Comment:
```suggestion
            if (account.balance < 6){
```
The account must have at least 6 Lunas in order to pay for all the extended transactions.

Reply:
Good catch!